src/Core/src/Sts/StsClient.php

```diff
@@ -89,9 +89,121 @@ public function getCallerIdentity($input = []): GetCallerIdentityResponse
         return new GetCallerIdentityResponse($response);
     }
 
-    protected function getEndpointPattern(?string $region): string
+    protected function getEndpointMetadata(string $region): array
     {
-        return $region ? parent::getEndpointPattern($region) : 'https://sts.amazonaws.com';
+        switch ($region) {
+            case 'af-south-1':
+            case 'ap-east-1':
+            case 'ap-northeast-1':
+            case 'ap-northeast-2':
+            case 'ap-south-1':
+            case 'ap-southeast-1':
+            case 'ap-southeast-2':
+            case 'ca-central-1':
+            case 'eu-central-1':
+            case 'eu-north-1':
+            case 'eu-south-1':
+            case 'eu-west-1':
+            case 'eu-west-2':
+            case 'eu-west-3':
+            case 'me-south-1':
+            case 'sa-east-1':
+            case 'us-east-1':
+            case 'us-east-2':
+            case 'us-west-1':
+            case 'us-west-2':
+                return [
+                    'endpoint' => 'https://sts.%region%.amazonaws.com',
```

Comment:
reading https://docs.aws.amazon.com/IAM/latest/UserGuide/programming.html#IAMEndpoints, it seems safer to use the global endpoint?

Reply:
meh, I guess it's better to respect what the user requests.

Reply:
Agreed, if the user choose a region, it should be used.

But because, configuration have a default value for region, user will always have a default region. I wonder, if `us-east-1` is enabled by default for STS.

If not, the fix would be to allow null region in this method, and take `null` as `global`. 
WDYT

Reply:
oh yes, i didn't notice you made the region non nullable. 
I think it has to remain nullable to use the global endpoint, yes.

Reply:
or at least provide some way of using the global endpoint, like you say

Reply:
config would have to be like this:

```yaml
async_aws:
    config:
        region: us-east-1

    clients:
		sts:
            region: ~
		iam:
            region: ~
```

Reply:
IMO no need to reset STS's region to null. You can not activate a service for a region while keeping another disabled.
If you setup `region: us-east-1` in the default config, it means, you have activated the region, and STS will be activated to.

The single edge-case I see is using `new StsClient([])` when `us-east-1` is not enabled. This can occurs when using `IniFileProvider` with a profile pointing to a Role

note: IAM is not impacted, as it does not have regional endpoint, and all region will fallback to global endpoint.

Reply:
PR updated:
* `new IamClient()` => global
* `new IamClient(['eu-west-1'])` => global
* `new IamClient(['us-east-1'])` => global
* `new StsClient()` => global
* `new StsClient(['eu-west-1'])` => eu-west-1
* `new StsClient(['us-east-1'])` => us-east-1
* `new SqsClient()` => us-east-1
* `new StsClient(['eu-west-1'])` => eu-west-1
* `new StsClient(['us-east-1'])` => us-east-1